source/common/quic/envoy_quic_server_session.cc

```diff
@@ -205,6 +206,9 @@ void EnvoyQuicServerSession::ProcessUdpPacket(const quic::QuicSocketAddress& sel
     maybeHandleCloseDuringInitialize();
   }
   quic::QuicServerSessionBase::ProcessUdpPacket(self_address, peer_address, packet);
+  if (self_address != connection()->self_address()) {
```

Comment:
Should we instead check that it's equal to the preferred address instead of that it's not self_address? (Or is that hard to do?)

Reply:
I don't feel strongly about this. Happy to see it land as-is.

Reply:
Given enough time to iterates on this PR, my QUICHE change to expose the SPA from QuicConnection gets into Envoy now. I changed this comparison to be equal to the preferred address.